diff --git a/CMakeLists.txt b/CMakeLists.txt
index 1e74325b68..d882701d5e 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -34,21 +34,16 @@ message(STATUS "")
 
 cmake_minimum_required(VERSION 3.13.4)
 #
-# We support all policy changes up to version 3.11.
+# We support all policy changes up to version 3.13.4
 #
 cmake_policy(VERSION 3.13.4)
 
-if(POLICY CMP0074)
-  # Introduced in CMake 3.12: find_package(<PackageName>) also searches
-  # <PackageName>_ROOT CMake and environment variables when set to NEW.
-  cmake_policy(SET CMP0074 NEW)
-endif()
-
-if(POLICY CMP0075)
-  # Introduced in CMake 3.12: Use CMAKE_REQUIRED_LIBRARIES also in include
-  # file checks. We set the policy to NEW explicitly in order to avoid
-  # spurious configure warnings.
-  cmake_policy(SET CMP0075 NEW)
+if(POLICY CMP0144)
+  # Introduced in CMake 3.27: find_package(<PackageName>) searches
+  # prefixes specified by upper-case <PACKAGENAME>_ROOT CMake and
+  # environment variables in addition to the case-preserved
+  # <PackageName>_ROOT variables.
+  cmake_policy(SET CMP0144 NEW)
 endif()
 
 if(POLICY CMP0167)
