diff --git a/examples/IBLevelSet/ex3/CMakeLists.txt b/examples/IBLevelSet/ex3/CMakeLists.txt
index a1114a6f2..8fab4f002 100644
--- a/examples/IBLevelSet/ex3/CMakeLists.txt
+++ b/examples/IBLevelSet/ex3/CMakeLists.txt
@@ -21,11 +21,13 @@ IBAMR_ADD_EXAMPLE(
   EXAMPLE_GROUP
     examples-IBLevelSet
   SOURCES
-    LevelSetInitialCondition.cpp  
-    SetFluidSolidDensity.cpp      
-    SetFluidSolidViscosity.cpp    
-    TagLSRefinementCells.cpp      
+    LevelSetInitialCondition.cpp
+    SetFluidSolidDensity.cpp
+    SetFluidSolidViscosity.cpp
+    TagLSRefinementCells.cpp
     example.cpp
+  REQUIRES
+    IBAMR_HAVE_LIBMESH
   LINK_TARGETS
     IBAMR2d
   INPUT_FILES
