diff --git a/lib/petsc/conf/rules b/lib/petsc/conf/rules
index 7219d0274e7..a24e64130c4 100644
--- a/lib/petsc/conf/rules
+++ b/lib/petsc/conf/rules
@@ -53,7 +53,21 @@ ${PETSC_DIR}/${PETSC_ARCH}/tests/testfiles:
 	@${MKDIR} -p ${PETSC_DIR}/${PETSC_ARCH}/tests && touch -t 197102020000 ${PETSC_DIR}/${PETSC_ARCH}/tests/testfiles
 
 libs: ${PETSC_DIR}/${PETSC_ARCH}/lib/petsc/conf/files ${PETSC_DIR}/${PETSC_ARCH}/tests/testfiles
-	+@cd ${PETSC_DIR} && MAKEFLAGS="-j$(MAKE_NP) -l$(MAKE_LOAD) $(MAKEFLAGS)" ${OMAKE_PRINTDIR} -f gmakefile ${MAKE_PAR_OUT_FLG} V=${V} libs
+	+@r=`echo "${MAKEFLAGS}" | grep ' \-j'`; \
+        if [ "$$?" = 0 ]; then \
+          make_j="";\
+          echo "Skipping adding -jMAKE_NP option as -j option is already specified"; \
+        else \
+          make_j="-j${MAKE_NP}"; \
+        fi; \
+	r=`echo "${MAKEFLAGS}" | grep ' \-l'`; \
+        if [ "$$?" = 0 ]; then \
+          make_l="";\
+          echo "Skipping adding -lMAKE_LOAD option as -l option is already specified"; \
+        else \
+          make_l="-l${MAKE_LOAD}"; \
+        fi; \
+        cd ${PETSC_DIR} && ${OMAKE_PRINTDIR} -f gmakefile $${make_j} $${make_l} ${MAKE_PAR_OUT_FLG} V=${V} libs
 
 # Does nothing; needed for some rules that require actions.
 foo:
diff --git a/makefile b/makefile
index a8329057970..8cdd2fc5af1 100644
--- a/makefile
+++ b/makefile
@@ -122,8 +122,8 @@ info:
 	-@echo "------------------------------------------"
 	-@echo "Using mpiexec: ${MPIEXEC}"
 	-@echo "------------------------------------------"
-	-@echo "Using MAKE: $(MAKE)"
-	-@echo "Using MAKEFLAGS: -j$(MAKE_NP) -l$(MAKE_LOAD) $(MAKEFLAGS)"
+	-@echo "Using MAKE: ${MAKE}"
+	-@echo "Default MAKEFLAGS: MAKE_NP:${MAKE_NP} MAKE_LOAD:${MAKE_LOAD} MAKEFLAGS:${MAKEFLAGS}"
 	-@echo "=========================================="
 
 #
-- 
2.25.1

